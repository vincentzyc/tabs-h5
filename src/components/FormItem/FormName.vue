<template>
  <div class="flex align-middle form-item">
    <div class="form-label">领卡姓名</div>
    <div class="form-input flex flex-auto align-middle">
      <input
        @blur="checkName(model)"
        :maxlength="15"
        placeholder="输入办理人姓名（已加密）"
        type="text"
        v-model.trim="model"
      />
    </div>
  </div>
</template>

<script lang="ts">
import { checkName } from '@/composition/business/use-verify-data'

export default defineComponent({
  name: "FormName",
  props: {
    modelValue: {
      required: true,
      type: String,
    }
  },
  emits: ['update:modelValue'],
  setup(props, { emit }) {
    const model = computed<string>({
      get() {
        return props.modelValue;
      },
      set(val) {
        emit("update:modelValue", val);
      }
    })
    return {
      model,
      checkName,
    }
  }
})
</script>