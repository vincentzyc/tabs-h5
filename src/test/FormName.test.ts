import { mount } from '@vue/test-utils'
import FormName from '@/components/FormItem/FormName.vue'

test('mount FormName component', async () => {
  expect(FormName).toBeTruthy()

  const wrapper = mount(FormName, {
    props: {
      modelValue: '',
      'onUpdate:modelValue': (e: string) => wrapper.setProps({ modelValue: e })
    }
  })

  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.get('input').setValue('测试姓名')

  expect(wrapper.props('modelValue')).toBe('测试姓名')

  const emitEvent = wrapper.emitted('update:modelValue')

  expect(emitEvent).toHaveLength(1)
})