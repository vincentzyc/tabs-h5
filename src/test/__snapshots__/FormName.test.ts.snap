// Vitest Snapshot v1

exports[`mount FormName component 1`] = `
"<div class=\\"flex align-middle form-item\\">
  <div class=\\"form-label\\">领卡姓名</div>
  <div class=\\"form-input flex flex-auto align-middle\\"><input maxlength=\\"15\\" placeholder=\\"输入办理人姓名（已加密）\\" type=\\"text\\"></div>
</div>"
`;
